### ISSUE TITLE
[0.9.5] InfluxDB throws 'panic: close of nil channel' if data directory is not writable

### ISSUE THREAD

**Issue by CVTJNII** [influxdb/influxdb#4946]
The following panic was seen on both 0.9.5.1 and nightly when the specified data directory was not writable:

```
 8888888           .d888 888                   8888888b.  888888b.                   
   888            d88P"  888                   888  "Y88b 888  "88b                  
   888            888    888                   888    888 888  .88P                  
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.                  
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b                 
   888   888  888 888    888 888  888   X88K   888    888 888    888                 
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P                 
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"                  
                                                                                     
2015/12/01 17:33:25 InfluxDB starting, version 0.9.5.1, branch 0.9.5, commit 9eab56311373ee6f788ae5dfc87e2240038f0eb4, built 2015-11-25T23:01:58+0000
2015/12/01 17:33:25 Go version go1.4.2, GOMAXPROCS set to 8                          
2015/12/01 17:33:25 no configuration provided, using default settings                
[metastore] 2015/12/01 17:33:25 Using data dir: /home/influxdb/.influxdb/meta        
[registration] 2015/12/01 17:33:25 registration service terminating                  
[retention] 2015/12/01 17:33:25 retention policy enforcement terminating             
[monitor] 2015/12/01 17:33:25 shutting down monitor system                           
[handoff] 2015/12/01 17:33:25 shutting down hh service                               
[subscriber] 2015/12/01 17:33:25 closed service                                      
panic: close of nil channel                                                          
                                                                                     
goroutine 1 [running]:                                                               
github.com/influxdb/influxdb/meta.(*localRaft).close(0xc20805f0e0, 0x0, 0x0)         
        /root/go/src/github.com/influxdb/influxdb/meta/state.go:206 +0x3e            
github.com/influxdb/influxdb/meta.(*Store).close(0xc20806c3c0, 0x0, 0x0)             
        /root/go/src/github.com/influxdb/influxdb/meta/store.go:557 +0x2fc           
github.com/influxdb/influxdb/meta.(*Store).Close(0xc20806c3c0, 0x0, 0x0)             
        /root/go/src/github.com/influxdb/influxdb/meta/store.go:519 +0x75            
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Close(0xc20808e120, 0x0, 0x0) 
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:472 +0x1e4
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open(0xc20808e120, 0x0, 0x0)  
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:427 +0x85
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run(0xc208068090, 0xc20800a000, 0x0, 0x0, 0x0, 0x0)
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:124 +0xe1b
main.(*Main).Run(0xc20802b940, 0xc20800a000, 0x0, 0x0, 0x0, 0x0)                     
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:86 +0x434      
main.main()                                                                          
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/main.go:46 +0xdc       
                                                                                     
goroutine 6 [syscall]:                                                               
os/signal.loop()                                                                     
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:21 +0x1f                 
created by os/signal.init·1                                                          
        /root/.gvm/gos/go1.4.2/src/os/signal/signal_unix.go:27 +0x35                 
                                                                                     
goroutine 17 [syscall, locked to thread]:                                            
runtime.goexit()                                                                     
        /root/.gvm/gos/go1.4.2/src/runtime/asm_amd64.s:2232 +0x1                     
                                                                                     
goroutine 8 [runnable]:                                                              
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc2080c6480, 0x7fa67491e1a8, 0xc20802c1c0, 0x0, 0x0)
        /root/go/src/github.com/influxdb/influxdb/tcp/mux.go:58 +0x15d               
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003                     
        /root/go/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:377 +0x963
```

This was traced down to the data directory not existing, or not being writable.  In 0.9.5 is looks like the default behavior is to try and write to ~/.influxdb instead of creating /var/opt/influxdb/.  (As an aside, as an admin, I much prefer having the default be /var/opt/influxdb.)  If the data directory does not exist and cannot be created the above error is thrown.

Please catch this case and fail gracefully.  This may trip users as the default data directory was changed.  Thanks.

**Comment by rossmcdonald** [comment]
@CVTJNII If you do not run InfluxDB with a configuration, then it will store data in your home directory (`~/.influxdb`) instead of the package default location (`/var/lib/influxdb`). If you installed using a package (deb, rpm), then all you should have to do is run `service influxdb start` in order to have InfluxDB use the `/var/lib/influxdb` directory.

I agree that the error is misleading, though, so that should definitely be fixed.

**Comment by CVTJNII** [comment]
Yes, the error is the spirit of this bug report.  I don't expect InfluxDB to start with an unwritable data directory, but the panic led me to believe this was a bug, not an environment problem.

If this could be caught with an error to tell the user InfluxDB can't start due to a data directory problem that would be great.

**Comment by oiooj** [comment]
fixed by #5016
